src/DataTypes/EnumValues.cpp

```diff
@@ -52,7 +55,23 @@ void EnumValues<T>::fillMaps()
 template <typename T>
 T EnumValues<T>::getValue(StringRef field_name, bool try_treat_as_id) const
 {
-    const auto it = name_to_value_map.find(field_name);
+    ContextPtr query_context;
+    if (CurrentThread::isInitialized())
+    {
+        query_context = CurrentThread::get().getGlobalContext();
+    }
+
+    auto it = name_to_value_map.find(field_name);
+    if (!it && query_context && query_context->getSettingsRef().cast_keys_to_string_from_json)
+    {
+        for (const auto& item : name_to_value_map)
+        {
+            if (item.getValue().second == std::stoi(field_name.toString()))
```

Comment:
Better use functions from ReadHelpers, such as `parse`.

Reply:
And you will avoid creating a temporary string object.

Reply:
This should be done only if there is no ambiguity.

Example: 

`Enum("2" = 1, "1" = 2, "Hello" = 3)`

In case of ambiguity, Enum values should be preferred instead of numeric indices.